### ISSUE TITLE
Cannot connect to hostname, IP works (HTTP)

### ISSUE THREAD

**Issue by yetdog** [standardnotes/syncing-server#40]
When trying to connect to a hostname, I get "Oops, a server error occurred when trying to sign in. Please try again." When I switch to the IP address that that hostname resolves to, sign in works.

Now, I would prefer to use HTTPS, but currently I'm attempting this over HTTP as I'm unclear about how to get an SSL certificate installed in the docker container.

This is leveraging the MacOS app as well as iOS; same behavior. Using docker-compose

**Comment by johnny243** [comment]
I would first check if the hostname is in fact resolving to the IP address:

`nslookup <your-hostname>`

You could manually edit your `/etc/hosts` file (on the host where you want to resolve the hostname) and specify it there.

**Comment by yetdog** [comment]
Thanks for the response. Confirmed resolution. Does hostname-based access require https perhaps?

On Mar 18, 2020, at 4:23 PM, Johnny Almonte <notifications@github.com> wrote:

﻿

I would first check if the hostname is in fact resolving to the IP address:

nslookup <your-hostname>

You could manually edit your /etc/hosts file (on the host where you want to resolve the hostname) and specify it there.

**Comment by johnny243** [comment]
> Thanks for the response. Confirmed resolution. Does hostname-based access require https perhaps?

I don't think so, it should connect via http just fine.

May I know which port did you choose for the `syncing-server`?

**Comment by yetdog** [comment]
3000.

On Mar 18, 2020, at 6:20 PM, Johnny Almonte <notifications@github.com> wrote:

﻿

Thanks for the response. Confirmed resolution. Does hostname-based access require https perhaps?

I don't think so, it should connect via http just fine.

May I know which port did you choose for the syncing-server?

**Comment by johnny243** [comment]
Then you should be able to see the default page when you access it. Example:

![image](https://user-images.githubusercontent.com/5891646/77022696-b6a0f480-6960-11ea-9970-a3ee2454b280.png)

Then in the Standard Notes app:

![image](https://user-images.githubusercontent.com/5891646/77022921-34650000-6961-11ea-9afa-ba1609bd5cc5.png)

![image](https://user-images.githubusercontent.com/5891646/77022964-4e064780-6961-11ea-933d-6771e6ef4791.png)

**Comment by yetdog** [comment]
Yep - works just fine pulling up the "Not supposed to be here" in a web browser. Simply can not sign in with iOS whatsoever. And logs...yeah. Docker *rules*

**Comment by yetdog** [comment]
Ok folks - problem identified but not solved. In iOS, you can *not* connect without HTTPS. Here are the logs from the app itself, pulled from my device:

""The resource could not be loaded because the App Transport Security policy requires the use of a secure connection.""

**Comment by yetdog** [comment]
I can open another issue on how to properly frontend this thing with https. Or would it just be a frontend nginx/letsencrypt box with its backend pointed at this system on port 3000?

**Comment by johnny243** [comment]
Sorry, I missed that you were using the iOS app, in which case it will not connect using a domain/subdomain unless it is HTTPS enabled. 

You either connect to it using the IP address or setup HTTPS on your `syncing-server`.

> I can open another issue on how to properly frontend this thing with https. Or would it just be a frontend nginx/letsencrypt box with its backend pointed at this system on port 3000?

You could setup a reverse proxy with `nginx` with HTTPS.
